new-log-viewer/src/contexts/StateContextProvider.tsx

```diff
@@ -240,17 +209,45 @@ const StateContextProvider = ({children}: StateContextProviderProps) => {
         handleMainWorkerResp,
     ]);
 
-    const loadPage = (newPageNum: number) => {
+    const loadPage = useCallback((
+        cursor: CursorType,
+    ) => {
         if (null === mainWorkerRef.current) {
             console.error("Unexpected null mainWorkerRef.current");
 
             return;
         }
+
         workerPostReq(mainWorkerRef.current, WORKER_REQ_CODE.LOAD_PAGE, {
-            cursor: {code: CURSOR_CODE.PAGE_NUM, args: {pageNum: newPageNum}},
+            cursor: cursor,
             decoderOptions: getConfig(CONFIG_KEY.DECODER_OPTIONS),
         });
-    };
+    }, []);
+
+    const loadPageAction = useCallback((
+        action: ACTION_NAME,
+        specificPageNum: Nullable<number> = null
```

Comment:
Instead of passing specificPageNum separately, can we create a type like CursorType that includes args? I know we already have `ActionType`, so we'll probably need to rename that to something like `UiActionType`.